File: src/main/java/com/github/tanokun/tanorpg/player/Attack.java
package com.github.tanokun.tanorpg.player;

public class Attack {
    private int attackCombo = 0;

    private int nextAttackCombo = 0;

    private int lastAttackTicks = 0;

    private boolean attackWait = false;

    public int getAttackCombo() {
        return attackCombo;
    }

    public int getNextAttackCombo() {
        return nextAttackCombo;
    }

    public int getLastAttackTicks() {
        return lastAttackTicks;
    }

    public boolean isAttackWait() {
        return attackWait;
    }

    public void setAttackCombo(int attackCombo) {
        this.attackCombo = attackCombo;
    }

    public void setNextAttackCombo(int nextAttackCombo) {
        this.nextAttackCombo = nextAttackCombo;
    }

    public int nextAttackCombo() {
        this.attackCombo++;
        return attackCombo;
    }

    public void setLastAttackTicks(int lastAttackTicks) {
        this.lastAttackTicks = lastAttackTicks;
    }

    public int nextLastAttackTicks() {
        this.lastAttackTicks++;
        return this.lastAttackTicks;
    }

    public void setAttackWait(boolean attackWait) {
        this.attackWait = attackWait;
    }
}

File: src/main/java/com/github/tanokun/tanorpg/player/Attack.kt
package com.github.tanokun.tanorpg.player

import com.github.tanokun.tanorpg.player.skill.SkillComboType

class Attack {
    var attackCombo = 0
    var nextAttackCombo = 0
    var lastAttackTicks = 0
    var isAttackWait = false
    var isSkillComboTrigger = false
    val skillCombos = ArrayList<SkillComboType>()
    var attack: Boolean = false

    fun nextAttackCombo(): Int {
        attackCombo++
        return attackCombo
    }

    fun nextLastAttackTicks(): Int {
        lastAttackTicks++
        return lastAttackTicks
    }
}